activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -28,9 +28,16 @@ module Associations
     # is computed directly through SQL and does not trigger by itself the
     # instantiation of the actual post records.
     class CollectionProxy < Relation
+      @@_extensions = {}
+
       def initialize(klass, association) #:nodoc:
         @association = association
         super klass, klass.arel_table, klass.predicate_builder
+
+        extensions = @@_extensions[association.reflection] ||=
+          association.method(:scope).super_method.().extensions
```

Comment:
The fact we need to call the super methods smell like we have something wrong in the inheritance chain. Can we refactor the code to avoid this?